test/org/zaproxy/zap/extension/pscanrulesBeta/ViewStateScannerTest.java

```diff
@@ -0,0 +1,197 @@
+
+		assertThat(alertsRaised.size(), equalTo(1));
+		assertThat(alertsRaised.get(0).getRisk(), equalTo(Alert.RISK_MEDIUM));
+		assertThat(alertsRaised.get(0).getConfidence(), equalTo(Alert.CONFIDENCE_MEDIUM));
+		assertThat(alertsRaised.get(0).getAlert(), equalTo("Emails Found in the Viewstate"));
+		assertThat(alertsRaised.get(0).getOtherInfo(), equalTo("[Itest@test.com]"));
+		assertThat(alertsRaised.get(0).getWascId(), equalTo(14));
+		assertThat(alertsRaised.get(0).getCweId(), equalTo(16));
+		assertSame(msg, alertsRaised.get(0).getMessage());
+	}	
+	
+	@Test
+	public void shouldRaiseAlertBecauseTheParametersDoesHaveIP() {
+		String encodedViewstate = getViewstateWithText("127.0.0.1");
+		msg.setResponseBody("<input name=\"__VIEWSTATE\" value=\"" + encodedViewstate +  "\">");
+		
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+		assertThat(alertsRaised.size(), equalTo(1));
+		assertThat(alertsRaised.get(0).getRisk(), equalTo(Alert.RISK_MEDIUM));
+		assertThat(alertsRaised.get(0).getConfidence(), equalTo(Alert.CONFIDENCE_MEDIUM));
+		assertThat(alertsRaised.get(0).getAlert(), equalTo("Potential IP Addresses Found in the Viewstate"));
+		assertThat(alertsRaised.get(0).getDescription(), equalTo("The following potential IP addresses were found being serialized in the viewstate field:"));
+		assertThat(alertsRaised.get(0).getOtherInfo(), equalTo("[127.0.0.1]"));
+		assertThat(alertsRaised.get(0).getWascId(), equalTo(14));
+		assertThat(alertsRaised.get(0).getCweId(), equalTo(16));
+		assertSame(msg, alertsRaised.get(0).getMessage());
+	}
+	
+	@Test
+	public void shouldRaiseAsViewstateIsSplit() {
+		msg.setResponseBody(	"<input type=\"hidden\" name=\"__VIEWSTATEFIELDCOUNT\" id=\"__VIEWSTATEFIELDCOUNT\" value=\"3\" />"
+					+ "<input type=\"hidden\" name=\"__VIEWSTATE\" id=\"__VIEWSTATE\" value=\"/wEPDwUKLTk2Njk3OTQxNg9kFgICAw9kFgICCQ88\" />" 
+					+ "<input type=\"hidden\" name=\"__VIEWSTATE1\" id=\"__VIEWSTATE1\" value=\"KwANAGQYAQUJR3JpZFZpZXcxD2dk4sjERFfnDXV/\" />"
+					+ "<input type=\"hidden\" name=\"__VIEWSTATE2\" id=\"__VIEWSTATE2\" value=\"hMFGAL10HQUnZbk=\" />");
+		
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+		assertThat(alertsRaised.size(), equalTo(2));
+		assertThat(alertsRaised.get(1).getRisk(), equalTo(Alert.RISK_INFO));
+		assertThat(alertsRaised.get(1).getAlert(), equalTo("Split Viewstate in Use"));
+		assertThat(alertsRaised.get(1).getWascId(), equalTo(14));
+		assertThat(alertsRaised.get(1).getCweId(), equalTo(16));
+		assertSame(msg, alertsRaised.get(0).getMessage());
+	}	
+		
+	/**
+	 * Helper function for injecten a piece of text into a valid base64 encoded ASP Viewstate. 
+	 * @param inject the string to inject
+	 * @return a base64 encoded string with the inject value injected at byte 40.
+	 */
+	private String getViewstateWithText(String inject) {
+		String base = "/wEPDwUJODczNjQ5OTk0D2QWAgIDD2QWAgIFDw8WAh4EVGV4dAUWSSBMb3ZlIERvdG5ldEN1cnJ5LmNvbWRkZMHbBY9JqBTvB5/6kXnY15AUSAwa";
```

Comment:
That was my first approach but the first part of the string is treated as a byte and has no-readable characters in it so it will not survive encoding / decoding if I start with a string.  

I might be possible to build a viewstate byte stream from scratch but do not have the knowledge. I was already hard to find info on what a good viewstate actually is. Let alone one which has email addresses or ip's.

So I came up with solution to be able to trigger these alerts.